### PR TITLE
Added legend unit test

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Trying out a bit of unit testing, I think all of the tests in this spec add value.

### CODE REVIEW

File: tests/chart/legendSpec.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    ['open', function(d) { return d.open; }],
+                    ['close', function(d) { return d.close; }]
+                ]);
+
+            var datum = {
+                open: 45.67,
+                close: 56.78
+            };
+
+            d3.select(element).data([datum])
```
Comment (chrisprice) [inline tests/chart/legendSpec.js]: Minor point: missing newline and I've been preferring datum for cases like this (single node selection).

File: tests/chart/legendSpec.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            // the second invocation should have an index of one
+            expect(labelSpy.calls.all()[1].args).toEqual([datum, 1]);
+            expect(valueSpy.calls.all()[1].args).toEqual([datum, 1]);
+
+            // the context should be the DOM element
+            expect(labelSpy.calls.all()[1].object.nodeName).toEqual('TH');
+            expect(valueSpy.calls.all()[1].object.nodeName).toEqual('TD');
+        });
+
+        it('should invoke the decorate unction with the correct parameters and context', function() {
```
Comment (chrisprice) [inline tests/chart/legendSpec.js]: Typo

File: tests/chart/legendSpec.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                .items([
+                    [labelSpy, valueSpy],
+                    [labelSpy, valueSpy]
+                ]);
+
+            var datum = {
+                open: 45.67
+            };
+
+            d3.select(element).data([datum])
```
Comment (chrisprice) [inline tests/chart/legendSpec.js]: As above

File: tests/chart/legendSpec.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                .rowDecorate(decorateSpy)
+                .items([
+                    ['header', 'value']
+                ]);
+
+            var datum = {
+                open: 45.67
+            };
+
+            d3.select(element).data([datum])
```
Comment (chrisprice) [inline tests/chart/legendSpec.js]: As above

**Comment by chrisprice** [comment]
Otherwise another *legend*ary PR...

**Comment by ColinEberhardt** [comment]
Thanks for the feedback

**Pull request merged**